OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -746,6 +746,14 @@ void INotifyBecomingIdle.OnBecomingIdle(Actor self)
 
 		protected virtual void OnBecomingIdle(Actor self)
 		{
+			// Add ammo check before other idle behaviors
+			if (rearmable != null && rearmable.Info.RearmActors.Count > 0 &&
+				self.TraitsImplementing<AmmoPool>().Any(p => !p.HasAmmo))
+			{
+				self.QueueActivity(new ReturnToBase(self));
+				return;
+			}
+
```

Comment:
The idle behavior shouldn't be changed. Players use planes to scout and to provide vision even if they are out of ammo. Besides we already have this behavior implemented, it's toggled in yaml. See `IdleBehaviorType.ReturnToBase` below.